apis/Google.Cloud.Diagnostics.Common/Google.Cloud.Diagnostics.Common/Trace/SimpleManagedTracer.cs

```diff
@@ -246,23 +246,24 @@ private void AnnotateSpan(TraceSpan span, Dictionary<string, string> labels)
         public void SetStackTrace(StackTrace stackTrace)
         {
             GaxPreconditions.CheckNotNull(stackTrace, nameof(stackTrace));
-            var currentStack = TraceStack;
-            CheckStackNotEmpty(currentStack);
-
-            AnnotateSpan(currentStack.Peek().TraceSpan, TraceLabels.FromStackTrace(stackTrace));
+            AnnotateSpan(TraceLabels.FromStackTrace(stackTrace));
         }
 
         /// <inheritdoc />
-        public string GetCurrentTraceId() =>_trace.TraceId;
+        public string GetCurrentTraceId() => _trace.TraceId;
         
         /// <inheritdoc />
         public ulong? GetCurrentSpanId() => GetCurrentSpanId(TraceStack);
```

Comment:
As mentioned in my comment below, make it the caller's responsibility to update `TraceStack`, so this would change to
```csharp
var currentStack = GetUpdatedStack(TraceStack);
var result = GetCurrentSpanId(currentStack);
TraceStack = currentStack;
return result;
```